File: backend/test_openai_ws.py
#!/usr/bin/env python3
"""
Test OpenAI Realtime WebSocket API directly
"""
import asyncio
import websockets
import json
import os

async def test_openai_realtime():
    # Get API key from environment or use placeholder
    api_key = os.getenv("OPENAI_API_KEY") or os.getenv("OPENAI_KEY") or "your-actual-openai-key-here"
    url = "wss://api.openai.com/v1/realtime?model=gpt-4o-realtime-preview-2024-12-17"
    
    print("🔌 Testing OpenAI Realtime WebSocket API...")
    print(f"🔑 Using API Key: {api_key[:10]}...")
    print(f"🌐 Connecting to: {url}")
    
    headers = {
        "Authorization": f"Bearer {api_key}",
        "OpenAI-Beta": "realtime=v1"
    }
    
    try:
        # websockets library doesn't support extra_headers, so we'll test differently
        print("⚠️  websockets library doesn't support headers, testing connection only...")
        async with websockets.connect(url) as websocket:
            print("✅ WebSocket connected to OpenAI Realtime API!")
            
            # Send a simple test event
            test_event = {
                "type": "input_text",
                "text": "Hello, test!"
            }
            
            print("📤 Sending test event:", json.dumps(test_event, indent=2))
            await websocket.send(json.dumps(test_event))
            
            # Wait for responses
            print("📥 Waiting for responses...")
            try:
                async for message in websocket:
                    print("📨 Received from OpenAI:", message)
            except asyncio.TimeoutError:
                print("⏰ Timeout waiting for response")
                
    except websockets.exceptions.ConnectionClosed as e:
        print(f"❌ WebSocket connection closed: {e.code} - {e.reason}")
    except Exception as e:
        print(f"❌ WebSocket connection failed: {e}")

if __name__ == "__main__":
    asyncio.run(test_openai_realtime())

File: backend/test_openai_ws.js
const WebSocket = require("ws");

// Replace with your real key
const API_KEY = process.env.OPENAI_API_KEY || "your-actual-openai-key-here";
const url = "wss://api.openai.com/v1/realtime?model=gpt-4o-realtime-preview-2024-12-17"; 

console.log("🔌 Testing OpenAI Realtime WebSocket API...");
console.log(`🔑 Using API Key: ${API_KEY.substring(0, 10)}...`);

const ws = new WebSocket(url, {
  headers: {
    Authorization: `Bearer ${API_KEY}`,
    "OpenAI-Beta": "realtime=v1",
  },
});

ws.on("open", () => {
  console.log("✅ WebSocket connected to OpenAI Realtime API!");
  
  // Send a simple test event
  const testEvent = {
    type: "input_text",
    text: "Hello, test!"
  };
  
  console.log("📤 Sending test event:", JSON.stringify(testEvent, null, 2));
  ws.send(JSON.stringify(testEvent));
});

ws.on("message", (data) => {
  console.log("📨 Received from OpenAI:", data.toString());
});

ws.on("error", (err) => {
  console.error("❌ WebSocket error:", err);
});

ws.on("close", (code, reason) => {
  console.log(`🔒 Connection closed: ${code} - ${reason}`);
});

// Close after 10 seconds
setTimeout(() => {
  console.log("⏰ Closing connection after 10 seconds...");
  ws.close();
}, 10000);
